EventLog.md

```diff
@@ -0,0 +1,397 @@
+
+Now add to this complexity the fact that new instruments will generate multiple (maybe even an *unknown* number of) datasets per sequence step.  Either the familiar dataset label has to change or else we have to give up on the idea of mapping datasets to individual sequence steps via the dataset label itself.  I'll assume the later and so a query will have to be performed if we want to map a label to a particular step in a particular sequence of an observation.  (Or perhaps I'm wrong?  Maybe all he python scripts work with the filename format which doesn't reference the dataset label at all?)
+
+In order to "perform a query" a dataset table obviously has to exist in the model.  I'll assume that the seqexec will be able to determine which datasets are associated with the step it just asked the instrument to observe.  These can be written into a dataset table containing columns for dataset id, sequence id, step number, creation time, filename, etc.
+
+Regardless, a sequence id concept is necessary.  It isn't clear if acquisition vs. science are the only kinds of sequences that need to be supported, whether there can be multiple science sequences, etc.  If we could enumerate the kinds of sequences we could tighten the format of sequence ids and prevent errors. I'll assume for now though an open `String` that can contain anything except hyphen characters, `-`, since they are field separators in our labels.
+
+````scala
+package gem
+
+import scalaz._
+import Scalaz._
+
+object Sequence {
+
+  trait Id {
+    def oid: Observation.Id
+    def name: String
+  }
+
+  object Id extends ((Observation.Id, String) => Option[Sequence.Id]) {
+    def apply(o: Observation.Id, n: String): Option[Sequence.Id] =
+      !(n.contains('-') || "" === n.trim) option new Sequence.Id {
+        override val oid: Observation.Id = o
+        override val name: String        = n.trim
+
+        override def toString: String =
+          s"${oid.toString}-$n"
+      }
+
+    def fromString(s: String): Option[Sequence.Id] =
+      s.lastIndexOf('-') match {
+        case -1 => None
+        case  n =>
+          val (a, b) = s.splitAt(n)
+          Observation.Id.fromString(a).flatMap { Id(_, b.drop(1)) }
+      }
+
+    def unsafeFromString(s: String): Sequence.Id =
+      fromString(s) | sys.error(s"Malformed Sequence.Id: $s")
+
+    def unapply(arg: Id): Option[(Observation.Id, String)] =
+      Some((arg.oid, arg.name))
+  }
+
+}
+````
+
+## Observation Execution Events
+
+In today's OCS, the tcc and seqexec send observation execution events to an intermediary process called the "wdba" as they slew to targets and execute sequences.  Perhaps in the future these tools will write directly to the database?  At any rate, all the events share a timestamp and an observation id but some types of events need additional information.  In the multi-sequence future, the observation id should be replaced with a sequence id for all (or maybe just most) events.
```

Comment:
If the tcc/seqexec would write updates directly to the db, how would clients be notified. E.g. you may want to display on the OT that an observation has completed. Perhaps they would instead notify the OCS and it will update the db and notify cilents